HW0/P4/P4.py

```diff
@@ -40,38 +48,132 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_meas = np.transpose(np.mat(np.loadtxt('P4_measurements.txt', delimiter = ",")))
+        #Load the measurements text file into a matrix
+
+    stretch_matrix = np.matrix([[1/rx, 0, 0], [0, 1/ry, 0], [0, 0, 1/rz]])
+        # Matrix with the stretching factors
+
+    corrected = stretch_matrix * s_meas
+        # Correct position by multiplying the stretch matrix by the measured position
+ 
+    x_coords = np.array(corrected)[0, :]  # extract x coordinates
+    y_coords = np.array(corrected)[1, :]  # extract y coordinates
+    z_coords = np.array(corrected)[2, :]  # extract z coordinates
+    
+
+    ax.plot(x_coords, y_coords, z_coords,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
-
+    # Propagation Matrix
+    A = np.matrix([[1, 0, 0,         dt,          0,          0],
+                   [0, 1, 0,          0,         dt,          0],
+                   [0, 0, 1,          0,          0,         dt],
+                   [0, 0, 0, (1 - c*dt),          0,          0],
+                   [0, 0, 0,          0, (1 - c*dt),          0],
+                   [0, 0, 0,          0,          0, (1 - c*dt)]])
+    
```

Comment:
It might be easier to create the A matrix in the following way:

 A = np.matrix(np.zeros([6,6]))
np.fill_diagonal(A, [1,1,1,1 - c*dt,1 - c*dt,1 - c*dt])

A[0,3] = dt
A[1,4] = dt
A[2,5] = dt